internal/controllers/machinedeployment/machinedeployment_controller.go

```diff
@@ -255,33 +257,34 @@ func (r *Reconciler) getMachineSetsForDeployment(ctx context.Context, d *cluster
 	// List all MachineSets to find those we own but that no longer match our selector.
 	machineSets := &clusterv1.MachineSetList{}
 	if err := r.Client.List(ctx, machineSets, client.InNamespace(d.Namespace)); err != nil {
-		return nil, err
+		return nil, errors.Wrapf(err, "failed to list MachineSets")
 	}
 
 	filtered := make([]*clusterv1.MachineSet, 0, len(machineSets.Items))
 	for idx := range machineSets.Items {
 		ms := &machineSets.Items[idx]
-		log.WithValues("MachineSet", klog.KObj(ms))
+		log = log.WithValues("MachineSet", klog.KObj(ms))
 		selector, err := metav1.LabelSelectorAsSelector(&d.Spec.Selector)
 		if err != nil {
-			log.Error(err, "Skipping MachineSet, failed to get label selector from spec selector")
+			log.Error(err, "Skipping MachineSet, failed to get label selector from spec selector", "MachineSet", klog.KObj(ms))
 			continue
 		}
 
 		// If a MachineDeployment with a nil or empty selector creeps in, it should match nothing, not everything.
 		if selector.Empty() {
-			log.Info("Skipping MachineSet as the selector is empty")
+			log.V(4).Info("Skipping MachineSet as the selector is empty", "MachineSet", klog.KObj(ms))
```

Comment:
Note: I am not sure if verbosity level here should be that high/lower or not needed at all, it is for now just aligned with https://github.com/kubernetes-sigs/cluster-api/blob/e7cbd42383f1df3aea404c9eff058202b46dbfbf/cmd/clusterctl/client/alpha/machinedeployment.go#L141

Reply:
4 or 5 seems okay for me given: https://cluster-api.sigs.k8s.io/developer/logging.html

> Logs at higher levels of verbosity (>=4) are meant to document “how it happened”, providing insight on thorny parts of the code; a person reading those logs usually has deep knowledge of the codebase.
> Don’t use verbosity higher than 5.

@fabriziopandini WDYT 4 or 5 for logs like this, I would probably use 5?

Reply:
I used 5 for now, please let me know if we have to change it otherwise.